fixes rendering issue when dashboard contains multiple panels.

jjaakola opened the pull request:
#5889

Multi-panel dashboard rendering waits for profiler.ts:renderingCompleted
to set the this.$rootScope.panelsRendered to have value greater or equal
to number of panels in the dashboard.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=5890) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=5890) before we can accept your contribution.<br/><hr/>**Jarkko Jaakola** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

jjaakola committed:
fixes rendering issue when dashboard contains multiple panels.

Multi-panel dashboard rendering waits for profiler.ts:renderingCompleted
to set the this.$rootScope.panelsRendered to have value greater or equal
to number of panels in the dashboard.

torkelo commented:
the render api is only built for single panel rendering currently, https://github.com/grafana/grafana/issues/2085

IldarMinaev commented on vendor/phantomjs/render.js:
```diff
@@ -55,7 +55,8 @@
         var rootScope = body.injector().get('$rootScope');
         if (!rootScope) {return false;}
-        return rootScope.panelsRendered;
+        var panelsToLoad = window.angular.element('div.panel').length;
```
It must be like this:
var panelsToLoad = window.angular.element('div.panel').length - window.angular.element('div.panel-drop-zone').length;
                        return rootScope.performance.panelsRendered >= panelsToLoad;
Because some 'div.panel' elements not real panels.

torkelo replied:
? there is no panel-drop-zone element in the solo panel view
http://play.grafana.org/dashboard-solo/db/graph-styles?from=1473928585116&to=1473932185116&panelId=3&width=1000&height=500&fullscreen

IldarMinaev replied:
But 'panel-drop-zone' can be exist in whole dashboard. When rendering whole dashboard with all panels panelsToLoad=window.angular.element('div.panel').length return more than needed so render.js can stack here.

IldarMinaev replied:
Without checking window.angular.element('div.panel-drop-zone').length in case of rendering full dashboard with all panels, render.js ends only on 'tries=1000' count, with my code it returns faster than 'tries' count exceed limit.
